server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -1134,7 +1118,11 @@ private WildcardExpressionResolver() {
         public static Collection<String> resolve(Context context, List<String> expressions) {
             Objects.requireNonNull(expressions);
             if (context.getOptions().expandWildcardExpressions() == false) {
-                return expressions;
+                if (expressions.size() == 1 && expressions.iterator().next().equals(Metadata.ALL)) {
```

Comment:
`expressions` is a List, so you can just do `expressions.get(0)`

Reply:
Pushed [71a8a47](https://github.com/elastic/elasticsearch/pull/91735/commits/71a8a4779784780300d212582dbb7ca415a0f7e3)